plumbing/hash/object.go

```diff
@@ -0,0 +1,29 @@
+package hash
+
+import "io"
+
+// ObjectID represents a calculated hash, which is immutable by nature.
+type ObjectID interface {
```

Comment:
I'd rename this to `Hash` to keep naming consistent

Reply:
I am holding back on making this change while `plumbing.Hash` exists, to avoid additional confusion on having multiple "Hash" types. At present `plumbing.Hash` has more than one use which is leading me to think that potentially it could be split into two different types. Either way, let's make that decision together with the final design for the API (as per discord thread) - which will be in a follow-up PR.